pcsx2/PINE.cpp

```diff
@@ -123,11 +121,13 @@ PINEServer::PINEServer(SysCoreThread* vm, unsigned int slot)
 	// that a "reasonable" value is 5, which is not.
 	listen(m_sock, 4096);
 
-	// we save a handle of the main vm object
-	m_vm = vm;
+	// we allocate once buffers to not have to do mallocs for each IPC
+	// request, as malloc is expansive when we optimize for µs.
+	m_ret_buffer = new char[MAX_IPC_RETURN_SIZE];
```

Comment:
Use a `std::vector<u8>` or `std::unique_ptr<u8[]>` here, we're trying to avoid raw pointers in new code.

Reply:
I went with vector as much as I could, some `u8*` still exist unfortunately. Maybe in the future `std::span` could replace all those though

Reply:
For what it's worth, we do have gsl::span here (until we make C++20 a hard requirement, but that's difficult because of libc++ being behind).